subsystem/src/main/java/org/wildfly/extension/elytron/tls/subsystem/SSLContextDefinitions.java

```diff
@@ -0,0 +1,2031 @@
+            .setAllowedValues(ALLOWED_PROTOCOLS)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition WANT_CLIENT_AUTH = new SimpleAttributeDefinitionBuilder(Constants.WANT_CLIENT_AUTH, ModelType.BOOLEAN, true)
+            .setMinSize(1)
+            .setDefaultValue(ModelNode.FALSE)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition NEED_CLIENT_AUTH = new SimpleAttributeDefinitionBuilder(Constants.NEED_CLIENT_AUTH, ModelType.BOOLEAN, true)
+            .setMinSize(1)
+            .setDefaultValue(ModelNode.FALSE)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition AUTHENTICATION_OPTIONAL = new SimpleAttributeDefinitionBuilder(Constants.AUTHENTICATION_OPTIONAL, ModelType.BOOLEAN, true)
+            .setMinSize(1)
+            .setDefaultValue(ModelNode.FALSE)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition USE_CIPHER_SUITES_ORDER = new SimpleAttributeDefinitionBuilder(Constants.USE_CIPHER_SUITES_ORDER, ModelType.BOOLEAN, true)
+            .setMinSize(1)
+            .setDefaultValue(ModelNode.TRUE)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition MAXIMUM_SESSION_CACHE_SIZE = new SimpleAttributeDefinitionBuilder(Constants.MAXIMUM_SESSION_CACHE_SIZE, ModelType.INT, true)
+            .setValidator(new IntRangeValidator(-1))
+            .setDefaultValue(new ModelNode(-1))
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition SESSION_TIMEOUT = new SimpleAttributeDefinitionBuilder(Constants.SESSION_TIMEOUT, ModelType.INT, true)
+            .setValidator(new IntRangeValidator(-1))
+            .setDefaultValue(new ModelNode(-1))
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition WRAP = new SimpleAttributeDefinitionBuilder(Constants.WRAP, ModelType.BOOLEAN, true)
+            .setDefaultValue(ModelNode.FALSE)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition MAXIMUM_CERT_PATH = new SimpleAttributeDefinitionBuilder(Constants.MAXIMUM_CERT_PATH, ModelType.INT, true)
+            .setValidator(new IntRangeValidator(1))
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    @Deprecated
+    static final SimpleAttributeDefinition MAXIMUM_CERT_PATH_CRL = new SimpleAttributeDefinitionBuilder(Constants.MAXIMUM_CERT_PATH, ModelType.INT, true)
```

Comment:
Since this is a new subsystem, I think we can remove this to avoid adding something that's deprecated to start out with.

Reply:
Alright, will update code and the relevant schemas. If someone is moving from elytron to elytron-tls and still has this configuration, should it just be left to raise errors since its deprecated?

Reply:
Alright. Should I just drop this from the XML schema, or leave it in and raise an exception when someone attempts to use it?

Reply:
We should drop the deprecated part from the schema.

Because it's expected that the new subsystem might have different / more simple configuration, I think it's ok to not worry about trying to handle older Elytron config in the new subsystem. If someone still tries to specify an older attribute, they'll get an error anyway since it won't be part of the new schema.